conda_forge_metadata/repodata.py

```diff
@@ -0,0 +1,140 @@
+"""
+Utilities to deal with repodata
+"""
+
+import bz2
+import json
+import os
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from functools import lru_cache
+from itertools import product
+from logging import getLogger
+from pathlib import Path
+from typing import Any, Dict, Generator, Iterable, List, Union
+from urllib.request import urlretrieve
+
+import bs4
+import requests
+
+logger = getLogger(__name__)
+
+SUBDIRS = (
+    "linux-64",
+    "linux-aarch64",
+    "linux-ppc64le",
+    "osx-64",
+    "osx-arm64",
+    "win-64",
+    "win-arm64",
+    "noarch",
+)
+CACHE_DIR = Path(".repodata_cache")
+
+
+@lru_cache
+def all_labels():
+    if token := os.environ.get("BINSTAR_TOKEN"):
+        label_info = requests.get(
+            "https://api.anaconda.org/channels/conda-forge",
+            headers={"Authorization": f"token {token}"},
+        ).json()
+
+        return sorted(label for label in label_info if "/" not in label)
+    else:
+        logger.info(
```

Comment:
Oh wow. This is clever but for sure will break eventually.

Reply:
🤞 I guess we can use the token in the by-the-numbers repo?

Reply:
Let me add a test so we find out if it breaks.

Reply:
Yes. Also we could write a json blob of the labels there to make available via this package.

Reply:
That's an excellent idea!

Reply:
We should probably not put the token in the by the numbers repo for security reasons. However, we can run the slow thing there and then revisit the token thing if it breaks.